WordPress/Classes/ViewRelated/Me/Account Settings/AccountSettingsViewController.swift

```diff
@@ -45,24 +47,26 @@ private class AccountSettingsController: SettingsController {
     }
     private let alertHelper = DestructiveAlertHelper()
 
-    init(service: AccountSettingsService) {
-        self.service = service
+    init(accountSettingsService: AccountSettingsService,
+         accountService: AccountService = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)) {
+        self.accountSettingsService = accountSettingsService
+        self.accountService = accountService
```

Comment:
[nit] What do you think of making `AccountService` optional instead, so we can do something like this? We do this in a couple different places, so I thought it might be nice for the sake of consistency 😄 
```suggestion
    init(accountSettingsService: AccountSettingsService,
         accountService: AccountService? = nil) {
        self.accountSettingsService = accountSettingsService
        self.accountService = accountService ?? AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
```

Reply:
Hmm well I think this is misleading, since this would imply that the `AccountSettingsController` doesn't necessarily rely on `AccountService` (but it does), since it can be optional. And another thing, at the point of use one would expect that passing a `nil` as an argument would result in that property to be `nil`, which would not be the case. Let me know what you think.

Reply:
I think that having a non-optional `AccountService` as a property for `AccountSettingsController` implies that it's a dependency. 

@leandroalonso Do you have any insight / comments about the `xyzService? = nil` [dependency injection pattern ](https://github.com/wordpress-mobile/WordPress-iOS/search?p=1&q=%22Service%3F+%3D+nil%22+path%3A%2FWordPress%2FClasses) we use throughout the repo?

Reply:
> I think that having a non-optional AccountService as a property for AccountSettingsController implies that it's a dependency.
I agree, but I don't feel having an optional parameter for a non optional property is the right thing.

Reply:
While it would be good to follow the pattern, I agree with @nikola-milicevic to make the dependency explicit (ps.: I've done the `nil` a few times myself but looking back it doesn't seem the best decision. Shame! 😝)

I think we can keep it that way (non-nil) and just remember to apply this same pattern to future code.

Reply:
@momo-ozawa @leandroalonso thanks for the comments! I'm merging it now as is.